[focus-without-user-activation] Update explainer (#584)
w3c/webappsec-feature-policy: policies/focus-without-user-activation.md

# `focus-without-user-activation` Permissions Policy

## Participate

- [Original Issue: w3c/webappsec-permissions-policy#273](https://github.com/w3c/webappsec-permissions-policy/issues/273)

> - [GitHub Issues on WHATWG HTML](https://github.com/whatwg/html/issues?q=focus-without-user-activation)

Reviewer: ```suggestion
- [GitHub Issues on WHATWG HTML](https://github.com/whatwg/html/issues?q=%22focus-without-user-activation%22)
```
Needs to be quoted or pulls in everything mentioning "focus", "without", etc

Author: Oh yes, good catch.

        }
      }, 100);
    });
    ```

    This approach is unreliable: it creates visible focus flickering, races with legitimate user interactions (what if the user intentionally clicked the iframe?), and doesn't scale across multiple iframes.

## Goals

- Let embedders **block programmatic focus** from embedded iframes (ads, widgets, third-party apps).
- Support control via both **HTTP response headers** and the **iframe `allow` attribute**.
- **Preserve user-initiated focus.** Clicking into an iframe or tabbing with the keyboard must always work.
- Support **focus delegation**, where a focused parent frame explicitly passes focus to a child iframe.
- **Preserve focus behavior within the iframe**. When a frame already has focus (or a descendant does), it can still move focus within its own subtree, even if the policy is denied.

## Non-goals

- Blocking user-initiated focus (clicks, keyboard tabbing). This policy only restricts programmatic focus.
- Replacing the `sandbox` mechanism. This policy complements it.
- Restricting focus by default. The default allowlist is `'*'` (all origins allowed), so this is opt-in.

## Proposed Approach

### How it works

> When the `focus-without-user-activation` policy is **disabled** for a document:

Reviewer: I think it's clearer to say either the policy is **denied**, or the **feature** controlled by the policy is disabled.
Saying the policy itself is disabled is IMO potentially confusing -- could be interepreted as saying the opposite of what's intended.

- `autofocus` attributes are ignored (unless the element is inserted as a result of a user gesture)
- `element.focus()` calls have no effect unless triggered by user activation

> - `window.focus()` calls are blocked in the same way

Reviewer: I think it's clearer just to write it out.
```suggestion
- `window.focus()` calls have no effect unless triggered by user activation
```

> - `dialog.showModal()` and popover focusing are also restricted

Reviewer: ```suggestion
- `dialog.showModal()` and popover focusing are blocked unless triggered by user activation
```
